ci/docker/vmcontrol.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python3
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# -*- coding: utf-8 -*-
+"""Utilities to control a guest VM"""
+
+__author__ = 'Pedro Larroy'
+__version__ = '0.1'
+
+import os
+import sys
+import subprocess
+import argparse
+import logging
+from subprocess import call, check_call, Popen, DEVNULL, PIPE
+import time
+import sys
+import multiprocessing
+import shlex
+
+###################################################
+#
+# Virtual testing with QEMU
+#
+# We start QEMU instances that have a local port in the host redirected to the ssh port.
+#
+# The VMs are provisioned after boot, tests are run and then they are stopped
+#
+
+QEMU_RUN="""
```

Comment:
"?

Reply:
what do you mean?

Reply:
You have 3 ", but if you intended an empty string you would have 2.